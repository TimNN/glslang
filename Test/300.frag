#version 300 es

uniform sampler2D s2D;
uniform sampler3D s3D;
uniform samplerCube sCube;
uniform samplerCubeShadow sCubeShadow;
uniform sampler2DShadow s2DShadow;
uniform sampler2DArray s2DArray;
uniform sampler2DArrayShadow s2DArrayShadow;

uniform isampler2D is2D;
uniform isampler3D is3D;
uniform isamplerCube isCube;
uniform isampler2DArray is2DArray;

uniform usampler2D us2D;
uniform usampler3D us3D;
uniform usamplerCube usCube;
uniform usampler2DArray us2DArray;

in float c1D;
in vec2  c2D;
in vec3  c3D;
smooth vec4  c4D;

flat in int   ic1D;
flat in ivec2 ic2D;
flat in ivec3 ic3D;
flat in ivec4 ic4D;
noperspective in vec4 badv; // ERROR
in sampler2D bads; // ERROR
precision lowp uint;       // ERROR

struct s {
    int i;
    sampler2D s;   // ERROR
};

out s badout;      // ERROR

void main()
{
    float f;
    vec4 v;
    v = texture(s2D, c2D);
    v = textureProj(s3D, c4D);
    v = textureLod(s2DArray, c3D, 1.2);
    f = textureOffset(s2DShadow, c3D, ic2D, c1D);
    v = texelFetch(s3D, ic3D, ic1D);
    v = texelFetchOffset(s2D, ic2D, 4, ic2D);
    f = textureLodOffset(s2DShadow, c3D, c1D, ic2D);
    v = textureProjLodOffset(s2D, c3D, c1D, ic2D);
    v = textureGrad(sCube, c3D, c3D, c3D);
    f = textureGradOffset(s2DArrayShadow, c4D, c2D, c2D, ic2D);
    v = textureProjGrad(s3D, c4D, c3D, c3D);
    v = textureProjGradOffset(s2D, c3D, c2D, c2D, ic2D);

    ivec4 iv;
    iv = texture(is2D, c2D);
    iv = textureProjOffset(is2D, c4D, ic2D);
    iv = textureProjLod(is2D, c3D, c1D);
    iv = textureProjGrad(is2D, c3D, c2D, c2D);
    iv = texture(is3D, c3D, 4.2);
    iv = textureLod(isCube, c3D, c1D);
    iv = texelFetch(is2DArray, ic3D, ic1D);

    iv.xy = textureSize(sCubeShadow, 2);

    float precise;
    double boo;       // ERROR
    dvec2 boo2;       // ERROR
    dvec3 boo3;       // ERROR
    dvec4 boo4;       // ERROR

    f += gl_FragCoord.y;
    gl_FragDepth = f;
}

float imageBuffer;    // ERROR, reserved
float uimage2DRect;   // ERROR, reserved
