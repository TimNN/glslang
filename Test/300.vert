#version 300 es

uniform mat4x3 m43;
uniform mat3x3 m33;
uniform mat4x4 m44;

in vec3 v3;
varying vec2 v2;               // ERROR, varying reserved
in vec4 bad[10];               // ERROR, no arrayed inputs
highp in vec4 badorder;        // ERROR, incorrect qualifier order
out invariant vec4 badorder2;  // ERROR, incorrect qualifier order
in centroid vec4 badorder4;    // ERROR, incorrect qualifier order
out flat vec4 badorder3;       // ERROR, incorrect qualifier order
void bar(in const float a);    // ERROR, incorrect qualifier order
void bar2(highp in float b);   // ERROR, incorrect qualifier order
smooth flat out vec4 rep;      // ERROR, replicating interpolation qualification
centroid sample out vec4 rep2; // ERROR, replicating auxiliary qualification
in uniform vec4 rep3;          // ERROR, replicating storage qualification

struct S {
    vec3 c;
    float f;
};

out S s;

void main()
{
    int id = gl_VertexID + gl_InstanceID;

    int c0 = gl_MaxVertexAttribs;
    int c1 = gl_MaxVertexUniformVectors;
    int c2 = gl_MaxVertexOutputVectors;
    int c3 = gl_MaxFragmentInputVectors;
    int c4 = gl_MaxVertexTextureImageUnits;
    int c5 = gl_MaxCombinedTextureImageUnits;
    int c6 = gl_MaxTextureImageUnits;
    int c7 = gl_MaxFragmentUniformVectors;
    int c8 = gl_MaxDrawBuffers;
    int c9 = gl_MinProgramTexelOffset;
    int c10 = gl_MaxProgramTexelOffset;

    mat3x4 tm = transpose(m43);
    highp float dm = determinant(m44);
    mat3x3 im = inverse(m33);

    mat3x2 op = outerProduct(v2, v3);

    gl_Position = m44[2];
    gl_PointSize = v2.y;

     s.c = v3;
     s.f = dm;

#ifdef GL_ES
#error GL_ES is set
#else
#error GL_ES is not set
#endif
}

float badsize[];    // ERROR
float[] badsize2;   // ERROR
uniform ub {
    int a[];        // ERROR
} ubInst[];         // ERROR
void foo(int a[]);  // ERROR
float okayA[] = float[](3.0, 4.0);  // Okay

out vec3 newV;
void newVFun()
{
    newV = v3;
}

invariant newV;  // ERROR, variable already used
in vec4 invIn;
invariant invIn; // ERROR, in v300
out S s2;
invariant s2;
invariant out S s3;
flat out int;
