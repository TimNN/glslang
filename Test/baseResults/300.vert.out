ERROR: 0:8: 'varying' : Reserved word. 
ERROR: 0:8: 'varying' : no longer supported in es profile; removed in version 300
ERROR: 0:9: 'vertex input arrays' : not supported with this profile: es
ERROR: 0:10: '' : precision qualifier must appear as last qualifier 
ERROR: 0:11: '' : invariant qualifier must appear first 
ERROR: 0:12: '' : Auxiliary qualifiers (centroid, patch, and sample) must appear before storage and precision qualifiers 
ERROR: 0:12: '' : vertex input cannot be further qualified 
ERROR: 0:13: '' : interpolation qualifiers must appear before storage and precision qualifiers 
ERROR: 0:14: '' : in/out must appear before const 
ERROR: 0:15: '' : precision qualifier must appear as last qualifier 
ERROR: 0:16: '' : can only have one interpolation qualifier (flat, smooth, noperspective) 
ERROR: 0:17: 'sample' : Reserved word. 
ERROR: 0:17: '' : can only have one auxiliary qualifier (centroid, patch, and sample) 
ERROR: 0:18: 'uniform' : too many storage qualifiers 
ERROR: 0:56: '#error' : GL_ES is set  
ERROR: 0:62: '' : array size required 
ERROR: 0:63: '' : array size required 
ERROR: 0:65: '' : array size required 
ERROR: 0:64: '' : array size required 
ERROR: 0:67: '' : array size required 
ERROR: 0:78: 'invariant' : can only apply to an output: invIn
ERROR: 0:88: 'ub2' : Cannot reuse block name within the same interface: uniform
ERROR: 0:92: 'ub2' : Cannot reuse block name within the same interface: uniform
ERROR: 0:96: 'ub2' : Cannot reuse block name within the same interface: uniform
ERROR: 0:104: 'ub3' : Cannot reuse block name within the same interface: uniform
ERROR: 25 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:27  Function Definition: main( (void)
0:27    Function Parameters: 
0:29    Sequence
0:29      Sequence
0:29        move second child to first child (highp int)
0:29          'id' (highp int)
0:29          add (highp int)
0:29            'gl_VertexID' (gl_VertexId highp int)
0:29            'gl_InstanceID' (gl_InstanceId highp int)
0:31      Sequence
0:31        move second child to first child (highp int)
0:31          'c0' (highp int)
0:31          Constant:
0:31            64 (const int)
0:32      Sequence
0:32        move second child to first child (highp int)
0:32          'c1' (highp int)
0:32          Constant:
0:32            128 (const int)
0:33      Sequence
0:33        move second child to first child (highp int)
0:33          'c2' (highp int)
0:33          Constant:
0:33            16 (const int)
0:34      Sequence
0:34        move second child to first child (highp int)
0:34          'c3' (highp int)
0:34          Constant:
0:34            15 (const int)
0:35      Sequence
0:35        move second child to first child (highp int)
0:35          'c4' (highp int)
0:35          Constant:
0:35            32 (const int)
0:36      Sequence
0:36        move second child to first child (highp int)
0:36          'c5' (highp int)
0:36          Constant:
0:36            32 (const int)
0:37      Sequence
0:37        move second child to first child (highp int)
0:37          'c6' (highp int)
0:37          Constant:
0:37            32 (const int)
0:38      Sequence
0:38        move second child to first child (highp int)
0:38          'c7' (highp int)
0:38          Constant:
0:38            16 (const int)
0:39      Sequence
0:39        move second child to first child (highp int)
0:39          'c8' (highp int)
0:39          Constant:
0:39            32 (const int)
0:40      Sequence
0:40        move second child to first child (highp int)
0:40          'c9' (highp int)
0:40          Constant:
0:40            -8 (const int)
0:41      Sequence
0:41        move second child to first child (highp int)
0:41          'c10' (highp int)
0:41          Constant:
0:41            7 (const int)
0:43      Sequence
0:43        move second child to first child (highp 3X4 matrix of float)
0:43          'tm' (highp 3X4 matrix of float)
0:43          transpose (highp 3X4 matrix of float)
0:43            'm43' (uniform highp 4X3 matrix of float)
0:44      Sequence
0:44        move second child to first child (highp float)
0:44          'dm' (highp float)
0:44          determinant (highp float)
0:44            'm44' (uniform highp 4X4 matrix of float)
0:45      Sequence
0:45        move second child to first child (highp 3X3 matrix of float)
0:45          'im' (highp 3X3 matrix of float)
0:45          inverse (highp 3X3 matrix of float)
0:45            'm33' (uniform highp 3X3 matrix of float)
0:47      Sequence
0:47        move second child to first child (highp 3X2 matrix of float)
0:47          'op' (highp 3X2 matrix of float)
0:47          outer product (highp 3X2 matrix of float)
0:47            'v2' (smooth out highp 2-component vector of float)
0:47            'v3' (in highp 3-component vector of float)
0:49      move second child to first child (highp 4-component vector of float)
0:49        'gl_Position' (gl_Position highp 4-component vector of float)
0:49        direct index (highp 4-component vector of float)
0:49          'm44' (uniform highp 4X4 matrix of float)
0:49          Constant:
0:49            2 (const int)
0:50      move second child to first child (highp float)
0:50        'gl_PointSize' (gl_PointSize highp float)
0:50        direct index (highp float)
0:50          'v2' (smooth out highp 2-component vector of float)
0:50          Constant:
0:50            1 (const int)
0:52      move second child to first child (highp 3-component vector of float)
0:52        c: direct index for structure (highp 3-component vector of float)
0:52          's' (smooth out structure{c,f})
0:52          Constant:
0:52            0 (const int)
0:52        'v3' (in highp 3-component vector of float)
0:53      move second child to first child (highp float)
0:53        f: direct index for structure (highp float)
0:53          's' (smooth out structure{c,f})
0:53          Constant:
0:53            1 (const int)
0:53        'dm' (highp float)
0:68  Sequence
0:68    move second child to first child (2-element array of highp float)
0:68      'okayA' (2-element array of highp float)
0:68      Constant:
0:68        3.000000
0:68        4.000000
0:71  Function Definition: newVFun( (void)
0:71    Function Parameters: 
0:73    Sequence
0:73      move second child to first child (highp 3-component vector of float)
0:73        'newV' (smooth out highp 3-component vector of float)
0:73        'v3' (in highp 3-component vector of float)
0:?   Linker Objects
0:?     'm43' (uniform highp 4X3 matrix of float)
0:?     'm33' (uniform highp 3X3 matrix of float)
0:?     'm44' (uniform highp 4X4 matrix of float)
0:?     'v3' (in highp 3-component vector of float)
0:?     'v2' (smooth out highp 2-component vector of float)
0:?     'bad' (in 10-element array of highp 4-component vector of float)
0:?     'badorder' (in highp 4-component vector of float)
0:?     'badorder2' (invariant smooth out highp 4-component vector of float)
0:?     'badorder4' (centroid in highp 4-component vector of float)
0:?     'badorder3' (flat out highp 4-component vector of float)
0:?     'rep' (smooth flat out highp 4-component vector of float)
0:?     'rep2' (centroid smooth sample out highp 4-component vector of float)
0:?     'rep3' (in highp 4-component vector of float)
0:?     's' (smooth out structure{c,f})
0:?     'badsize' (unsized array of highp float)
0:?     'badsize2' (unsized array of highp float)
0:?     'ubInst' (layout(column_major shared ) uniform unsized array of block{a})
0:?     'okayA' (2-element array of highp float)
0:?     'newV' (smooth out highp 3-component vector of float)
0:?     'invIn' (in highp 4-component vector of float)
0:?     's2' (smooth out structure{c,f})
0:?     's3' (invariant smooth out structure{c,f})
0:?     'a' (layout(column_major shared ) uniform block{f})
0:?     '__anon__0' (layout(column_major shared ) uniform block{b23})
0:?     'gl_VertexID' (gl_VertexId highp int)
0:?     'gl_InstanceID' (gl_InstanceId highp int)

